src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoFilter } from '../../types/TodoFilter';
+
+type Props = {
+  todos: Todo[],
+  todoFilter: TodoFilter,
+  setTodoFilter: (filter: TodoFilter) => void,
+  deleteTodosCompleted: (completedId: number[]) => void,
+  completedId: number[],
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  todoFilter,
+  setTodoFilter,
+  deleteTodosCompleted,
+  completedId,
+}) => {
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${itemsLeft} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: todoFilter === TodoFilter.ALL },
+          )}
+          onClick={() => setTodoFilter(TodoFilter.ALL)}
```

Comment:
Create separate handler instead of using anonymous function.
Fix all similar cases